pypeit/edgetrace.py

```diff
@@ -3970,326 +3988,235 @@ def fully_masked_traces(self, flag=None, exclude=None):
-        if _design_file is None or not os.path.isfile(_design_file):
-            msgs.error('Slit-mask design file not found or none provided.')
-
-        # Get the paramters to use
-        maxiter = self.par['mask_reg_maxiter']
-        maxsep = self.par['mask_reg_maxsep']
-        sigma = self.par['mask_reg_sigrej']
-        ignore_alignment = self.par['ignore_alignment']
-
-        # TODO: Set allow_resync and design_file to be a parameters, as
-        # well?
-
-        # Read the design data
-        msgs.info('Reading slit-mask design information from: {0}'.format(_design_file))
-        if self.spectrograph.get_slitmask(_design_file) is None:
-            msgs.error('Unable to read design file or no slit-mask design reader '
-                       'defined for {0}.'.format(self.spectrograph.spectrograph))
-
-        # Match both left and right edges simultaneously
-        x_design = np.array([self.spectrograph.slitmask.bottom[:,0],
-                             self.spectrograph.slitmask.top[:,0]]).T.ravel()
-        reference_row = self.left_pca.reference_row if self.par['left_right_pca'] \
-                            else self.pca.reference_row
-        x_det = self.edge_fit[reference_row,:]
-
+        # `traceimg` must have knowledge of the flat frame that built it
+        if self.spectrograph.get_slitmask(self.traceimg.files[0]) is None:
+            msgs.error('Unable to read slitmask design info')
+        if self.spectrograph.get_grating(self.traceimg.files[0]) is None:
+            msgs.error('Unable to read grating info')
+        if self.spectrograph.get_amapbmap(self.traceimg.files[0]) is None:
+            msgs.error('Unable to read amap and bmap')
+
+        # Match left and right edges separately
+        # Position of the slits in mm from the slit-mask design
+        xcen_mm = self.spectrograph.slitmask.center[:, 0]
+        sortindx = np.argsort(xcen_mm)
+
+        xtop_mm = self.spectrograph.slitmask.top[:, 0]
+        xbot_mm = self.spectrograph.slitmask.bottom[:, 0]
+
+        ytop_mm = self.spectrograph.slitmask.top[:, 1]
+        ybot_mm = self.spectrograph.slitmask.bottom[:, 1]
+
+        slitindx = self.spectrograph.slitmask.slitindx
+        slit_id = self.spectrograph.slitmask.slitid
+
+        npoints = 250
+        nslits = self.spectrograph.slitmask.nslits
+        wave = np.arange(npoints) * 24. + 4000.
+
+        # Left (bottom) and right (top) traces in pixels from optical model
+        # bottom
+        xb_img, yb_img, ccd_b, xb_pix, yb_pix = self.spectrograph.mask_to_pixel_coordinates(x=xbot_mm, y=ybot_mm,
+                                                                                            wave=wave)
+        # top
+        xt_img, yt_img, ccd_t, xt_pix, yt_pix = self.spectrograph.mask_to_pixel_coordinates(x=xtop_mm, y=ytop_mm,
+                                                                                            wave=wave)
+
```

Comment:
Does taking the median perform better than just evaluating the trace at the midpoint spectral position?

Reply:
I haven't really investigated that.